content/docs/secure/encryption/rekey.md

```diff
@@ -0,0 +1,142 @@
+weight: 4
+Description: >
+  Rekey Configuration and Usage
+---
+
+## Rekey Controller Installation
+
+The CSM Encryption Rekey CRD Controller is an optional component that, if installed, allows encrypted volumes rekeying in a
+Kubernetes cluster. The Rekey Controller can be installed via the standard Dell Helm Chart repository available 
+at https://github.com/dell/helm-charts.
+
+A secret of the cluster config must be created with the name ``cluster-kube-config`` typically from the .kube/config. Here is an example:  
+
+```shell
+ kubectl create secret generic cluster-kube-config --from-file=/home/root/.kube/config
+```
+
+The Rekey Controller Helm chart defines these values:
+
+
+```yaml
+# Rekey controller image name.
+image: dellemc/csm-encryption-rekey-controller:v0.1.0
+
+# Rekey controller image pull policy.
+# Allowed values:
+#  Always: Always pull the image.
+#  IfNotPresent: Only pull the image if it does not already exist on the node.
+#  Never: Never pull the image.
+imagePullPolicy: IfNotPresent
+
+# logLevel: Log level of the rekey controller.
+# Allowed values: "error", "warning", "info", "debug", "trace".
+logLevel: "info"
+
+# This value is required and must match encryption.pluginName value
+# of the corresponding Dell CSI driver.
+provisioner:
+
+# This value is required and must match encryption.apiPort value
+# of the corresponding Dell CSI driver.
+port:
+```
+
+| Parameter | Description | Required | Default |
+| --------- |-------------|----------|--|
+| image | Rekey controller image name. | No | dellemc/csm-encryption-rekey-controller:v0.1.0 |
+| imagePullPolicy | Rekey controller image pull policy. | No | IfNotPresent |
+| logLevel | Log level of the rekey controller. | No | info |
+| provisioner | This value is required and must match `encryption.pluginName` value of the corresponding Dell CSI driver. | Yes |  |
+| port | This value is required and must match `encryption.apiPort` value of the corresponding Dell CSI driver. | Yes |  |
+
+## Deploy Rekey Controller
+
+Copy the values.yaml to a local file. Once the CSM Encryption Rekey Controller local values.yaml file has been
+adjusted for the current cluster, deploy the controller by installing the Helm chart. As an example:
+
+``` helm install --values local-values.yaml rekey-controller csm-encryption-rekey-controller```
+
+A rekey-controller pod should now be up and running.
+
+
+## Rekey Controller Usage
+
+The general procedure for rekeying is to create a rekey custom resource via a simple yaml configuration. This 
+will kick off a rekey process on the PV specified as the `volume`in the resource. 
+
+### Rekey with dellctl
+
+If dellctl CLI is installed, rekey'ing a volume is simple. First, identify a volume with a PV that is encrypted with the CSM Encryption driver
+provisioner.
+
+For example, let's rekey an encrypted PV with the name ``k8s-112a5d41bc``, and call our rekey object `myrekey`:
+
+```shell
+$ dellctl encryption rekey myrekey k8s-112a5d41bc

```

Comment:
These commands can also be added to the CLI docs at `content/docs/references/cli/`

Reply:
done. please review